mypy/build.py

```diff
@@ -2342,11 +2342,15 @@ def dispatch(sources: List[BuildSource], manager: BuildManager) -> Graph:
         # after we loaded cache for whole graph. The `read_protocol_cache` will also validate
         # the protocol cache against the loaded individual cache files.
         TypeState.proto_deps = read_protocol_cache(manager, graph)
-        if TypeState.proto_deps is None and manager.stats.get('fresh_trees', 0) > 0:
+        if TypeState.proto_deps is None and manager.stats.get('fresh_metas', 0) > 0:
             # There were some cache files read, but no protocol dependencies loaded.
-            manager.errors.set_file(_cache_dir_prefix(manager), None)
-            manager.errors.report(0, 0, "Error reading protocol dependencies cache -- aborting",
-                                  blocker=True)
+            manager.log("Error reading protocol dependencies cache -- aborting cache load")
+            manager.cache_enabled = False
+            TypeState.proto_deps = {}
+            manager.log("Falling back to full run -- reloading graph...")
+            graph = load_graph(sources, manager)
```

Comment:
Should we call `update_protocol_deps` as well? This might be less likely to get stale if we'd only have one `process_graph` callin this function. Maybe instead of calling `load_graph` and `process_graph` here, call `dispatch` recursively after changing the options?

Reply:
> Should we call `update_protocol_deps` as well?

I think no, since `update.py` should do this. We call it manually below to have them up to date if we write caches. Anyway, I think calling `dispatch` recursively is a better option, so we don't need to worry about keeping the logic in sync.